cvpy/image/NaturalImageTable.py

```diff
@@ -1,21 +1,192 @@
+                   add_columns: List(str) = None, copy_vars: List(str) = None,
+                   output_table_parms: Dict[str,str] = {}) -> ImageTable:
+        """
+        Applies masking to an ImageTable.
+
+        Parameters
+        ------------
+        mask : :class:`cvpy.ImageTable`
+            Specifies the image table that will be used for masking.
+        decode : :class:`bool`
+            Specifies whether to decode the output image table.
+        add_columns : :class:'List(str)'
+            Specifies extra columns to add to the output table.
+        copy_vars : :class:`List(str)`
+            Specifies which columns to copy to the output image table.
+        output_table_parms : :class:`Dict[str,str]`
+            Specifies the parameters in the output image table.
+        Returns
+        ------------
+        :class:`cvpy.ImageTable`
+        """
+
+        conn = self._cas_session
+
+        ###############################################
+        ########### Mask Tbl Decoded ##################
+        ###############################################
+        if mask.has_decoded_images():
+
+            # List of columns that will be alternated
+            alter_columns = [dict(name=mask.image, rename="seg"),
+                             dict(name=mask.dimension, rename="dim"),
+                             dict(name=mask.resolution, rename="res"),
+                             dict(name=mask.imageFormat, rename="form")]
+
+            # SQL string to create the mask table
+            fed_sql_str = f'''create table _images_to_mask_ {{options replace=true}} as 
+                select a.seg, a.dim, a.res, a.form, b.* 
+                from {mask.table.name} as a right join {self.image.table.name} as b 
+                on a._id_=b._id_ '''
+
+            # Dictionary for specifying information in our binary operation
+            binary_operation_dict = dict(binaryOperationType="MASK_SPECIFIC",
+                                         image="seg", 
+                                         dimension="dim",
+                                         resolution="res", 
+                                         imageFormat="form")
+
+            # Mini method for renaming columns in ImageTable object
+            # *This step must be completed AFTER the alter table command*
+            def rename_columns():
+                mask.image = "seg"
+                mask.dimension = "dim"
+                mask.resolution = "res"
+                mask.imageFormat = "form"
+
+        ###############################################
+        ############# Mask Tbl Encoded ################
+        ###############################################
+        else:
+            # Create CAS Table
+            if 'name' not in output_table_parms:
+                output_table_parms['name'] = RandomNameGenerator().generate_name()
+            
+            cas_table = conn.CASTable(**output_table_parms)
```

Comment:
Since cas_table is required regardless of decoded images, I think you need to move creation of CAS Table above the "if" condition which starts on Line 105. Otherwise, this code will not work for decoded mask table.